src/plugins/timelist/TimelistComponent.vue

```diff
@@ -413,40 +423,47 @@ export default {
         return;
       }
 
-      const row = this.$el.querySelector('.js-list-item');
-      if (row && this.firstCurrentActivityIndex > -1) {
-        // scroll to somewhere mid-way of the current activities
-        const ROW_HEIGHT = row.getBoundingClientRect().height;
+      // See #7167 for scrolling algorithm
+      const scrollTop = this.calculateScrollOffset();
 
-        if (this.canAutoScroll() === false) {
-          return;
-        }
-
-        const scrollOffset =
-          this.currentActivitiesCount > 0 ? Math.floor(this.currentActivitiesCount / 2) : 0;
+      if (scrollTop === undefined) {
+        this.resetScroll();
+      } else {
         this.$el.parentElement?.scrollTo({
-          top: ROW_HEIGHT * (this.firstCurrentActivityIndex + scrollOffset),
+          top: scrollTop,
           behavior: 'smooth'
         });
         this.autoScrolled = false;
-      } else if (row && this.activityClosestToNowIndex > -1) {
-        // scroll to somewhere close to 'now'
+      }
+    },
+    calculateScrollOffset() {
+      let scrollTop;
 
+      //No scrolling necessary if no past events are present
+      if (this.pastActivitiesCount > 0) {
+        const row = this.$el.querySelector('.js-list-item');
```

Comment:
We should use a ref here

Reply:
I don't think there is a way to access a child component's refs. This is why I did it using the className.

Reply:
There's a few different ways to accomplish this, but after digging in a bit and based on the structure of the Timelist and ListView components I think this is actually permissible here.